src/server/collections/postsDatabase.js

```diff
@@ -60,11 +61,37 @@ function insertDocument(db, postInfo, callback) {
   });
 }
 
+function likePost(id, userName, callback) {
+  MongoClient.connect(url, (err, db) => {
+    if (err !== null) {
+      console.log('Couldn\'t get connect to the db', err);
+      return;
+    }
+    db.collection('posts').find({'_id': ObjectId(id)}).toArray(function(err, element) {
+      if(element[0].likes.includes(userName)) {
```

Comment:
these to findAndModifies are the same. you can call it once and just manage the new like arrays within the if and else